src/System.ComponentModel.EventBasedAsync/tests/BackgroundWorkerTests.cs

```diff
@@ -214,6 +214,42 @@ public void RunWorkerAsyncTwice()
         }
 
         [Fact]
+        public void TestCancelInsideDoWork()
+        {
+            var orignal = SynchronizationContext.Current;
+            try
+            {
+                var bw = new BackgroundWorker() { WorkerSupportsCancellation = true };
+                var barrier = new Barrier(2);
+
+                bw.DoWork += (sender, e) =>
+                {
+                    barrier.SignalAndWait();
+                    barrier.SignalAndWait();
+
+                    if (bw.CancellationPending)
+                    {
+                        e.Cancel = true;
+                    }
+                };
+
+                bw.RunWorkerCompleted += (sender, e) =>
+                {
+                    Assert.True(e.Cancelled);
+                };
+
+                bw.RunWorkerAsync();
+                barrier.SignalAndWait();
+                bw.CancelAsync();
+                barrier.SignalAndWait();
+            }
```

Comment:
Shouldn't we wait for the background operation to complete here?  Otherwise we may not actually execute the Assert.True(e.Cancelled), if the tests were to end prior to that executing.